### ISSUE TITLE
cargo run fails

### ISSUE THREAD

**Issue by jtbarker** [gruberb/web-programming-in-rust#1]
cargo build
   Compiling futures-core-preview v0.3.0-alpha.13
   Compiling httparse v1.3.3
   Compiling memchr v2.2.0
   Compiling num-integer v0.1.39
   Compiling serde v1.0.90
   Compiling slog v2.4.1
   Compiling ryu v0.2.7
   Compiling unsafe-any v0.4.2
error[E0554]: #![feature] may not be used on the stable release channel
 --> /Users/jonbarker/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.13/src/lib.rs:3:1
  |
3 | #![feature(futures_api)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0053]: method `poll` has an incompatible type for trait
  --> /Users/jonbarker/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.13/src/future/future_obj.rs:64:5
   |
64 |     fn poll(self: Pin<&mut Self>, waker: &Waker) -> Poll<T> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ types differ in mutability
   |
   = note: expected type `fn(std::pin::Pin<&mut future::future_obj::LocalFutureObj<'a, T>>, &mut std::task::Context<'_>) -> std::task::Poll<T>`
              found type `fn(std::pin::Pin<&mut future::future_obj::LocalFutureObj<'a, T>>, &std::task::Waker) -> std::task::Poll<T>`

error[E0053]: method `poll` has an incompatible type for trait
   --> /Users/jonbarker/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-preview-0.3.0-alpha.13/src/future/future_obj.rs:114:5
    |
114 |     fn poll(self: Pin<&mut Self>, waker: &Waker) -> Poll<T> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ types differ in mutability
    |
    = note: expected type `fn(std::pin::Pin<&mut future::future_obj::FutureObj<'a, T>>, &mut std::task::Context<'_>) -> std::task::Poll<T>`
               found type `fn(std::pin::Pin<&mut future::future_obj::FutureObj<'a, T>>, &std::task::Waker) -> std::task::Poll<T>`

error: aborting due to 3 previous errors

Some errors occurred: E0053, E0554.
For more information about an error, try `rustc --explain E0053`.
error: Could not compile `futures-core-preview`.
warning: build failed, waiting for other jobs to finish...
error: build failed

**Comment by peter-lyons-kehl** [comment]
Yes, several parts are out of date: they don't need `nightly` and its `#![feature(...)]` anymore (nor `02-x/heroku/RustConfig`). And it needs to be updated.
